bin/deployables

```diff
@@ -83,600 +99,829 @@ region=${DEPLOY_AWS_REGION}
+  exit 1
 }
 
-function ecs_verify_deploy () {
 
-    local subfamily="$1"
+##
+## public functions
+##
 
-    if [ -z "$subfamily" ] ; then
-        family="${DEPLOY_APP_NAME}"
-        service="${DEPLOY_APP_NAME}"
-    else
-        family="${DEPLOY_APP_NAME}-${subfamily}"
-        service="${DEPLOY_APP_NAME}-${subfamily}"
-    fi
+function docker_build {
 
-    # part two of some bash craziness to receive the revision_target
-    key="revision_target_$( echo $family | tr -s - _ )"
-    revision_target=${!key}
+  echo
+  echo Building docker image
+  echo DEPLOY_DOCKERFILE: ${DEPLOY_DOCKERFILE}
+  echo DEPLOY_DOCKER_LOCAL_TAG: ${DEPLOY_DOCKER_LOCAL_TAG}
+  echo
 
-    echo
-    echo Verifying deploy
-    echo revision_target: $revision_target
-    echo
+  if [ -z "${DEPLOY_DOCKERFILE}" ] ; then
+    echo "Error: DEPLOY_DOCKERFILE is required"
+    return 1
+  fi
 
-    # wait for older revisions to disappear
-    for attempt in {1..60}; do
+  if [ -z "${DEPLOY_DOCKER_LOCAL_TAG}" ] ; then
+    echo "Error: DEPLOY_DOCKER_LOCAL_TAG is required"
+    return 1
+  fi
 
-        revision_deployed=$( \
-            aws ecs describe-services \
-                --profile ${DEPLOY_AWS_PROFILE} \
-                --cluster ${DEPLOY_ECS_CLUSTER_NAME} \
-                --services ${service} \
-                | $JQ ".services[0].deployments | .[] | select(.taskDefinition != \"$revision_target\") | .taskDefinition" \
-            )
+  if [[ ! -z "${DEPLOY_GITHUB_MACHINE_USER_KEY_FINGERPRINT}" ]] ; then
 
-        if [ -z "$revision_deployed" ] ; then
-            echo "Success: Deployed!"
-            return 0
-        fi
+    fingerprint="$( echo ${DEPLOY_GITHUB_MACHINE_USER_KEY_FINGERPRINT} | tr -d : )"
 
-        echo "Waiting for update:"
-        echo "  revision_target: ${revision_target}"
-        echo "  revision_deployed: ${revision_deployed}"
-        sleep 5
+    eval keyfile="~/.ssh/id_${fingerprint}"
 
-    done
+    if [[ ! -f ${keyfile} ]] ; then
+      echo "Error: Unable to find machine user key file"
+      echo "- fingerprint: ${fingerprint}"
+      echo "- keyfile: ${keyfile}"
+
+      eval keyfile="~/.ssh/id_circleci_github"
+
+      if [[ ! -f ${keyfile} ]] ; then
+        echo "Error: Unable to find circle github key file"
+        echo "- keyfile: ${keyfile}"
+        return 1
+      fi
+
+    fi
+
+    echo "Using GITHUB_MACHINE_USER_KEY=${keyfile}"
+
+    docker build \
+      --rm=false \
+      --no-cache \
+      --build-arg GITHUB_MACHINE_USER_KEY="$(< ${keyfile} )" \
+      --build-arg NPM_TOKEN=$NPM_TOKEN \
+      -t ${DEPLOY_DOCKER_LOCAL_TAG} \
+      -f ${DEPLOY_DOCKERFILE} \
+      .
+
+  else
+
+    docker build \
+      --rm=false \
+      --build-arg NPM_TOKEN=$NPM_TOKEN \
+      -t ${DEPLOY_DOCKER_LOCAL_TAG} \
+      -f ${DEPLOY_DOCKERFILE} \
+      .
+
+  fi
 
-    echo "Error: Service update took too long."
-    exit 1
 }
 
 
-##
-## public functions
-##
+function _deploy_image () {
```

Comment:
Should this function be split up? It immediately forks into one of two branches and never meets back up afterwards, so it could potentially be 3 new functions: 2 for the 2 distinct code branches + 1 for the `DEPLOY_RETAG_AND_PUSH` check (and to keep other functions from having to make this check themselves)

Reply:
Alternatively, could the `DEPLOY_RETAG_AND_PUSH` check be used for deciding what the base tag (`$remote_tag` vs `DEPLOY_DOCKER_LOCAL_TAG`) and the `$target_tag` values should be and then just call `ecr_tag_image` and `ecr_push_image` for whatever those values are afterwards?

Maybe something like this?

```
if DEPLOY_RETAG_AND_PUSH is set {
  base_tag=...
  target_tag=...

  ecr_pull_image $base_tag
} else {
  base_tag=...
  target_tag=...
  
  if DEPLOY_PUSH_SECONDARY_TAG is set {
    secondary_tag=...
  }
}

ecr_tag_image $base_tag $target_tag
ecr_push_image $target_tag

if secondary_tag is set {
  ecr_tag_image $base_tag $secondary_tag
  ecr_push_image $secondary_tag
}
```

Reply:
We don't use `DEPLOY_RETAG_AND_PUSH` or `DEPLOY_PUSH_SECONDARY_TAG` anymore, so we should just delete that, simplifying this significantly.